spring/boot2-autoconfigure/src/main/java/com/linecorp/armeria/internal/spring/SpringDependencyInjector.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2022 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.internal.spring;
+
+import static java.util.Objects.requireNonNull;
+
+import org.springframework.beans.factory.BeanFactory;
+import org.springframework.beans.factory.NoSuchBeanDefinitionException;
+
+import com.google.common.base.MoreObjects;
+
+import com.linecorp.armeria.server.DependencyInjector;
+
+final class SpringDependencyInjector implements DependencyInjector {
+
+    static DependencyInjector of(BeanFactory beanFactory) {
+        requireNonNull(beanFactory, "beanFactory");
+        return new SpringDependencyInjector(beanFactory);
+    }
+
+    private final BeanFactory beanFactory;
+
+    private SpringDependencyInjector(BeanFactory beanFactory) {
+        this.beanFactory = beanFactory;
+    }
+
+    @Override
+    public <T> T getInstance(Class<T> type) {
+        try {
+            return beanFactory.getBean(type);
+        } catch (NoSuchBeanDefinitionException t) {
+            // Return null in order to apply the FallbackDependencyInjector.
+            return null;
```

Comment:
What do you think of warning or throwing for the specific case of `NoUniqueBeanDefinitionException`?
For this case, a spring bean is defined (actually multiple beans are) but the user will still encounter the generic exception message which isn't very helpful for this case.

Reply:
I never expected that `NoUniqueBeanDefinitionException` is the subclass of `NoSuchBeanDefinitionException`. 😅
I think we should rethrow `NoUniqueBeanDefinitionException`. Thanks. 😄